airbyte-cdk/bulk/core/load/src/main/kotlin/io/airbyte/cdk/task/DestinationTaskLauncher.kt

```diff
@@ -81,13 +87,53 @@ class DefaultDestinationTaskLauncher(
     private val processRecordsTaskFactory: ProcessRecordsTaskFactory,
     private val processBatchTaskFactory: ProcessBatchTaskFactory,
     private val closeStreamTaskFactory: CloseStreamTaskFactory,
-    private val teardownTaskFactory: TeardownTaskFactory
+    private val teardownTaskFactory: TeardownTaskFactory,
+    private val exceptionHandler: DestinationTaskLauncherExceptionHandler
 ) : DestinationTaskLauncher {
     private val log = KotlinLogging.logger {}
 
     private val runTeardownOnce = AtomicBoolean(false)
     private val batchUpdateLock = Mutex()
 
+    class SyncTaskWrapper(
+        private val exceptionHandler: DestinationTaskLauncherExceptionHandler,
+        private val innerTask: Task,
```

Comment:
nit: can this be `innerTask: SyncTask`?

and should this class implement SyncTask instead of raw Task?

Reply:
Interestingly, this forces StreamTaskWrapper to itself be a SyncTask, which is... what we want :)